### PR TITLE
 test: ceph: failures of applyDiffToRawImage

### PR DESCRIPTION
**Pull request by skoya76**
~Todo: merge this PR after https://github.com/cybozu-go/fin/pull/50 is merged~
Done

### CODE REVIEW

File: internal/infrastructure/ceph/rbd_test.go [hunk 1]
```diff
@@ -112,6 +113,120 @@ func TestZeroFill_success(t *testing.T) {
 	}
 }
 
+func TestApplyDiffToRawImage_error_InvalidHeader(t *testing.T) {
```
Comment (ushitora-anqou) [inline internal/infrastructure/ceph/rbd_test.go:116]: We need to write Arrange-Act-Assert comments and descriptions.

File: internal/infrastructure/ceph/rbd_test.go [hunk 2]
```diff
@@ -112,6 +113,120 @@ func TestZeroFill_success(t *testing.T) {
...
+	err = applyDiffToRawImage(getRawImagePathForTest(t), reader, "", "toSnap", 10)
+	assert.Error(t, err)
+}
+
+func TestApplyDiffToRawImage_error_MissingDiffFileName(t *testing.T) {
+	rbdRepository := NewRBDRepository()
+	err := rbdRepository.ApplyDiffToRawImage(
+		getRawImagePathForTest(t),
+		"non existing file",
+		"fromSnap",
```
Comment (ushitora-anqou) [inline internal/infrastructure/ceph/rbd_test.go:143]:
```suggestion
		"",
```

File: internal/infrastructure/ceph/rbd_test.go [hunk 3]
```diff
@@ -112,6 +113,120 @@ func TestZeroFill_success(t *testing.T) {
...
+}
+
+func TestApplyDiffToRawImage_error_MissingToSnapNameArg(t *testing.T) {
+	reader, err := diffgenerator.Run(
+		diffgenerator.WithToSnapName("toSnap"),
+		diffgenerator.WithImageSize(10),
+	)
+	require.NoError(t, err)
+
+	err = applyDiffToRawImage(getRawImagePathForTest(t), reader, "", "", 10)
```
Comment (ushitora-anqou) [inline internal/infrastructure/ceph/rbd_test.go:156]:
```suggestion
	err = applyDiffToRawImage(getRawImagePathForTest(t), reader, "", "" /* missing to-snap */, 10)
```

File: internal/infrastructure/ceph/rbd_test.go [hunk 4]
```diff
@@ -112,6 +113,120 @@ func TestZeroFill_success(t *testing.T) {
...
+}
+
+func TestApplyDiffToRawImage_error_ExpansionUnitSizeNonPositive(t *testing.T) {
+	reader, err := diffgenerator.Run(
+		diffgenerator.WithToSnapName("toSnap"),
+		diffgenerator.WithImageSize(10),
+	)
+	require.NoError(t, err)
+
+	err = applyDiffToRawImage(getRawImagePathForTest(t), reader, "", "toSnap", 0)
```
Comment (ushitora-anqou) [inline internal/infrastructure/ceph/rbd_test.go:167]:
```suggestion
	err = applyDiffToRawImage(getRawImagePathForTest(t), reader, "", "toSnap", 0 /* non positive expansion unit size */)
```

Comment (ushitora-anqou) [reply internal/infrastructure/ceph/rbd_test.go:167]: We should add test cases for negative sizes?

**Review by ushitora-anqou — CHANGES_REQUESTED** [review]

**Review by ushitora-anqou — APPROVED** [review]

**Review by llamerada-jp — APPROVED** [review]
